objectDetection_lite.py

```diff
@@ -0,0 +1,256 @@
+                index_j = sorted_indices[j]
+                if _bbox_iou(boxes[index_i], boxes[index_j]) >= nms_thresh:
+                    boxes[index_j].classes[c] = 0
+
+
+# load and prepare an image
+def _load_image_pixels(image, shape):
+
+    width, height, _ = np.shape(image)
+    image = cv2.resize(image, dsize=shape, interpolation=cv2.INTER_CUBIC)
+    # convert to numpy array
+    # scale pixel values to [0, 1]
+    image = image.astype('float32')
+    image /= 255.0
+    # add a dimension so that we have one sample
+    image = np.expand_dims(image, 0)
+    return image, width, height
+
+
+# get all of the results above a threshold
+def _get_boxes(boxes, labels, thresh):
+    v_boxes, v_labels, v_scores = list(), list(), list()
+    # enumerate all boxes
+    for box in boxes:
+        # enumerate all possible labels
+        for i in range(len(labels)):
+            # check if the threshold for this label is high enough
+            if box.classes[i] > thresh:
+                v_boxes.append(box)
+                v_labels.append(labels[i])
+                v_scores.append(box.classes[i] * 100)
+                # don't break, many labels may trigger for one box
+
+    return v_boxes, v_labels, v_scores
+
+
+def _object_position(width, height, cx, cy):
+
+    i, j = 0, 0
+    x_unit = cx / width
+    y_unit = cy / height
+    if 0 < x_unit < 1/3:
+        i = 0
+    elif 1/3 < x_unit < 2/3:
+        i = 1
+    elif 2/3 < x_unit < 1:
+        i = 2
+
+    if 0 < y_unit < 1/3:
+        j = 0
+    elif 1/3 < y_unit < 2/3:
+        j = 1
+    elif 2/3 < y_unit < 1:
+        j = 2
+
+    return i, j
+
+
+def init():
+
+    # define the labels
+    labels = ['شخص', 'دراجة', 'سيارة', 'دراجات نارية', 'طائرة', 'حافلة', 'قطار', 'شاحنة نقل', 'قارب',
+              'إشارة ضوئية', 'صنبور الاطفاء', 'لافتة توقف', 'عداد موقف السيارات', 'مقعد', 'طائر', 'قط', 'الكلب',
+              'حصان', 'خروف', 'بقرة', 'فيل', 'يتحمل', 'الحمار الوحشي', 'زرافة', 'حقيبة ظهر', 'مظلة', 'حقيبة يد',
+              'ربطة عنق', 'حقيبة سفر', 'الطبق الطائر', 'زحلوقة', 'لوح التزلج', 'الكرة الرياضية', 'طائرة ورقية',
+              'مضرب البيسبول', 'قفاز البيسبول', 'لوح تزلج', 'مزلجة', 'مضرب التنس', 'زجاجة', 'كأس نبيذ', 'كوب',
+              'فرع', 'سكين', 'ملعقة', 'عاء', 'موز', 'تفاحة', 'ساندويتش', 'البرتقالي', 'بروكلي', 'جزرة', 'نقانق',
+              'بيتزا', 'الدونات', 'كيك', 'كرسي', 'كنبة', 'النبات المحفوظ بوعاء', 'السرير', 'طاولة الطعام', 'الحمام',
+              'رصد التلفزيون', 'حاسوب محمول', 'الفأر', 'التحكم عن بعد', 'لوحة المفاتيح',
+              'الهاتف الخلوي', 'الميكروويف', 'فرن', 'محمصة خبز كهربائية', 'مكتب المدير', 'ثلاجة', 'كتاب', 'ساعة حائط',
+              'مزهرية', 'مقص', 'دمية دب', 'مجفف شعر', 'فرشاة الأسنان']
+
+    # Load TFLite model and allocate tensors.
+    interpreter = tf.lite.Interpreter(model_path="model2.tflite")
+    interpreter.allocate_tensors()
+
+    return interpreter, labels
+
+
+def detect(image, interpreter, labels):
```

Comment:
again, the caller shouldn't have the interpreter in the first place, your module should have it stored. this function should only take the image, and the labels.

Reply:
the problem is that the function that load model is the same function that return the reference to the object that use for detection

Reply:
I understand. Maybe store it globally and then access it in this function. Maybe give it an initial value of `None`.